GithubIssueTagger/Program.cs

```diff
@@ -11,86 +11,66 @@
 var allReportTypes = typeof(Program).Assembly.GetTypes().Where(t => t.IsClass && t.IsAssignableTo(typeof(IReport))).OrderBy(t => t.Name).ToList();
 
 var patOption = new Option<string>("--pat");
+patOption.Description = "GitHub Personal Access Token to make API calls with.";
 patOption.AddAlias("-p");
 
+var patEnvVarOption = new Option<string>("--pat-variable");
+patEnvVarOption.Description = "Environment variable that contains the GitHub Personal Access Token to use.";
```

Comment:
Does anyone have better ideas on how to provide either the PAT directly, or the environment variable name, other than two mutually exclusive options?

Maybe reuse the `--pat` option for both, but if a special prefix is used, then use environment variable instead? But the prefix would have to be something that is never a valid PAT.

Reply:
As far as I can tell, the most common convention seems to be that each tool hardcodes what environment variable is expects, so there is no CLI argument to tell it which arg to read. Pretty much all CI systems have ways to specify environment variables per step, so it doesn't matter if different tools in the same pipeline use different variable names for the same value.

As such, I plan on changing the code to look for a `GITHUB_TOKEN` environment variable, if the `--pat` argument isn't supplied. Arguably we shouldn't even support a `--pat` argument, so that secrets do not appear in command histories, even on your own local computer. Needing to set an env var before running a CLI tool is barely more effort than passing a `--pat` value, so I'm trying to decide whether or not to remove the `--pat` option altogether.

Reply:
I agree, just have the environment variable name hard coded and used if `--pat` is not specified.  I think the `--pat` command line argument is useful and isn't that much more secure than requiring a user to set the env var first.  You could also prompt for the password which is slightly more secure.

Reply:
Last year, [I added code to get the token stored in the git credential manager](https://github.com/NuGet/Entropy/pull/60/files) (ok, so I added it into a different tool, but then Nikolche copied it into this one), so as long as you can push to a NuGet repo (or clone a private NuGet repo), then the tool Just Works. Therefore, usually there's no need to provide a PAT at all.

@jeffkl, given this, do you think that `--pat` or interactively asking for the pat, is still useful?

Reply:
Well yeah shoot that's super useful if we can just make it work.  I just tried out `git credential fill`, I never knew about that.. 
 We should just run `git` and if it doesn't give us a PAT then just tell the user to authenticate with `git` first.  If someone complains, they can add a `--pat` command-line argument 😄